### PR TITLE
fix(JarResolver): AAG-2875: Unity jar resolver for iOS and Android

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
[AAG-2875]

## DESCRIPTION
Added the Unity jar resolver for iOS and Android with correct dependencies.

Instructions:

Drag and drop the unity package into your project.

*iOS:*
The iOS part of this was complete in a few hours, it just simply worked. To make this work all you need to do is resolve the libraries from: Top bar: Assets -> External Dependency Resolver -> iOS Resolver -> Install Cocoapods (This is installed in your unity editor folder)
Then set you player in build settings to iOS device and build the project.
Once the project builds you will need to set the provisioning profile and build it to your device.
That's it! No build flags to change anymore 🎉 

Improvements for the future:
- Moving the iOS Unity code to a SAUnity Cocoapod and away from this repo. 

*Android*
This took significantly longer to fix as there were dependency issues in the main sa-mobile-android-sdk repo.
To make this work, resolve the libraries Top bar: Assets -> External Dependency Resolver -> Android Resolver -> Resolve. This step may automatically run when you add the SuperAwesome Unity package to the project.
Then, set your player settings to Android and make sure that in the player settings you are building for arm64 or X86_64 depending on your chosen platform.
Build and run, you should be able to see ads. 🎉 

## SCREENSHOTS
n/a

[AAG-2875]: https://superawesomeltd.atlassian.net/browse/AAG-2875?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/2.build.unity.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  -gvh_disable \
+  -batchmode \
+  -importPackage "$(pwd)/source/external-dependency-manager-latest.unitypackage" \
+  -projectPath "$(pwd)/build/unity/$source/demo" \
+  -exportPackage \
+    "Assets/ExternalDependencyManager" \
+    "Assets/PlayServicesResolver" \
+    "Assets/Plugins" \
+    "Assets/SuperAwesome" \
+    "$(pwd)/output/SuperAwesomeSDK.$version.Unity.full.unitypackage" \
```
Comment (gunhansancar) [inline scripts/2.build.unity.sh:55]: Suggestion: We don't need `Unity.full` in the name for the output so we can update it here or in the consecutive PRs?

File: scripts/2.build.unity.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+projectProperties=build/unity/$source/demo/Assets/Plugins/Android/gradleTemplate.properties
+echo "target=android-11" >> $projectProperties
+echo "android.useAndroidX=true" >> $projectProperties
+echo "android.enableJetifier=true" >> $projectProperties
+
+# copy jar files
+cp build/android/unity-classes.jar output/unity-classes.jar || echo "No .jar files"
+
+# copy admob files
+cp -r build/android/admob/*.zip output
```
Comment (gunhansancar) [inline scripts/2.build.unity.sh:39]: Can we get `saadmob` from `jar-resolver`?

File: scripts/2.build.unity.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# step 2: add android resources
+############################################################
+
+projectProperties=build/unity/$source/demo/Assets/Plugins/Android/gradleTemplate.properties
+echo "target=android-11" >> $projectProperties
+echo "android.useAndroidX=true" >> $projectProperties
+echo "android.enableJetifier=true" >> $projectProperties
+
+# copy jar files
+cp build/android/unity-classes.jar output/unity-classes.jar || echo "No .jar files"
```
Comment (gunhansancar) [inline scripts/2.build.unity.sh:36]: Should we get rid of `unity-classes` or should we keep it?

**Review by gunhansancar — APPROVED** [review]

**Review by thomas-james-1986 — APPROVED** [review]
Nice!

**Pull request merged**